apps/lib/opt.c

```diff
@@ -1049,12 +1049,7 @@ char **opt_rest(void)
 /* How many items in remaining args? */
 int opt_num_rest(void)
 {
-    int i = 0;
-    char **pp;
-
-    for (pp = opt_rest(); *pp; pp++, i++)
-        continue;
-    return i;
+    return argc - opt_index - 1;
```

Comment:
`- 1` doesn't seem right.  When `argv[opt_index] == NULL`, then `opt_index == argc`, no?

Reply:
Why will `opt_index` ever get the value equal to `argc`?

Reply:
`opt_index == argc` when all the command line arguments have been parsed.  If your command was `openssl list -help`, this is what you should have:

- `argv[0] = "openssl"`
- `argv[1] = "list"`
- `argv[2] = "-help"`
- `argv[3] = NULL`
- `argc = 3`

When you have exhausted the command line, `opt_index` should be 3.  In this case, `opt_num_rest()` would return -1...  which isn't correct.

Reply:
> `opt_index == argc` when all the command line arguments have been parsed. If your command was `openssl list -help`, this is what you should have:
> 
> * `argv[0] = "openssl"`
> * `argv[1] = "list"`
> * `argv[2] = "-help"`
> * `argv[3] = NULL`
> * `argc = 3`
> 
> When you have exhausted the command line, `opt_index` should be 3. In this case, `opt_num_rest()` would return -1... which isn't correct.

No. In this case, `argc` should be 4 as `argv` contains 4 elements. No?

Reply:
**NO!!!**

Reply:
I suggest learning up about the C programming language and command line arguments.

Reply:
Ah, that's where the misunderstanding is!

For questions like these, https://cppreference.com/` is a pretty good source of truth:

https://en.cppreference.com/w/c/language/main_function

> **argc** - Non-negative value representing the number of arguments passed to the program from the environment in which the program is run.

Reply:
cppreference.com isn't an ideal reference here.  It's not C.
However, the C standard is the same here.

Reply:
Oh, and cppreference does talk about **argv** as well on the same page

Reply:
OK. Is this a better solution?
```
if (argc >= opt_index)
    return argc - opt_index - 1;
else
    return  argc - opt_index;
```